### PR TITLE
Create a Puppet Class for email-alert-api db-admin

### PR DESCRIPTION
**Pull request by suthagarht**
- This change creates a puppet class to manage email-alert-api db-admin
instances.

https://trello.com/c/ljsMBGQ4/1419-split-email-alert-api-into-its-own-rds-instance

Solo: @suthagarht

### CODE REVIEW

File: modules/govuk/manifests/node/s_email_alert_api_db_admin.pp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  # This class collects the resources that are exported by the
+  # govuk_postgresql::server::db defined type
+  include ::govuk_postgresql::server::not_slave
+
+  # Ensure the client class is installed
+  class { '::govuk_postgresql::client': } ->
+
+  # include all PostgreSQL classes that create databases and users
+  class { '::govuk::apps::content_performance_manager::db': }
```
Comment (issyl0) [inline modules/govuk/manifests/node/s_email_alert_api_db_admin.pp]: This is Email Alert API, not Warehouse.

File: modules/govuk/manifests/node/s_email_alert_api_db_admin.pp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  $postgres_backup_desc = 'RDS Warehouse PostgreSQL backup to S3'
```
Comment (issyl0) [inline modules/govuk/manifests/node/s_email_alert_api_db_admin.pp]: Same.

File: modules/govuk/manifests/node/s_email_alert_api_postgresql.pp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  $wale_private_gpg_key,
+  $wale_private_gpg_key_fingerprint,
+  $alert_hostname = 'alert.cluster',
+) inherits govuk::node::s_base  {
+  include govuk_env_sync
+  Govuk_mount['/var/lib/postgresql'] -> Class['govuk_postgresql::server::standalone']
+
+  include govuk_postgresql::server::standalone
+  include govuk_postgresql::tuning
+  include govuk::apps::content_performance_manager::db
```
Comment (issyl0) [inline modules/govuk/manifests/node/s_email_alert_api_postgresql.pp]: Wrong database.

File: modules/govuk/manifests/node/s_email_alert_api_db_admin.pp (hunk 1 above)
Comment (suthagarht) [reply modules/govuk/manifests/node/s_email_alert_api_db_admin.pp]: thank you, corrected

File: modules/govuk/manifests/node/s_email_alert_api_db_admin.pp (hunk 2 above)
Comment (suthagarht) [reply modules/govuk/manifests/node/s_email_alert_api_db_admin.pp]: thank you done

File: modules/govuk/manifests/node/s_email_alert_api_postgresql.pp (hunk 3 above)
Comment (suthagarht) [reply modules/govuk/manifests/node/s_email_alert_api_postgresql.pp]: thank you , done

**Comment by deanwilson** [comment]
We should either do a small refactor or add a dedup card to the backlog board to refactor the duplicate code out into a class/define when this work is done and deployed. the `gof3r` code for example should live in one place.

**Comment by issyl0** [comment]
@deanwilson Yep, this is duplicated across all the db-admin manifests so far.

**Comment by schmie** [comment]
Laura's solution to this was: https://github.com/alphagov/govuk-puppet/blob/b588e4ade996e97b8975e69cb00800521fff4a48/modules/hosts/manifests/purge.pp..
I am not sure, if there are a differences(?!) Should we unify this?

**Comment by issyl0** [comment]
Yeah, I pointed that solution out, but it still failed the tests.

**Pull request merged**